### ISSUE TITLE
FedNTD

### ISSUE THREAD

**Issue by jafermarq** [adap/flower#2013]
# FedNTD

* **Title:** Preservation of the Global Knowledge by Not-True Distillation in Federated Learning
* **Venue:** NeurIPS 2022
* **Link to paper:** https://papers.nips.cc/paper_files/paper/2022/hash/fadec8f2e65f181d777507d1df69b92f-Abstract-Conference.html

## Do you want to work on this baseline? 

> 🌻 Check everything about the `Summer of Reproducibility` on [`flower.dev/summer`](https://flower.dev/summer/)

> All available baselines are listed in the [Summer of Reproducibility Dashboard](https://github.com/orgs/adap/projects/6/) and also in the [GitHub Issues with the `summer-of-reproducibility` label](https://github.com/adap/flower/labels/summer-of-reproducibility). The content is the same.

> 📝 It is advised to complete these steps before your start working on your code. But if you can't wait to implement your baseline with Flower (we totally understand it 😄), please ensure you follow the steps on [how to contribute a new baseline](https://github.com/adap/flower/tree/main/baselines#contributing-a-new-baseline).

What follows are the steps 1 & 2 in the Summer of Reproducibility instructions.

### 1. Join the Summer of Reproducibility program
- [x] Join the [Flower Slack](https://flower.dev/join-slack/) and say _"hi! 👋"_ in the channel `#summer-of-reproducibility`.
- [x] Pick a baseline from our curated list **<---------------------------------------- [you are doing this now]**
### 2. Define the scope of your contribution
- [x] **What are you going to reproduce?** Add a comment to your issue and tell us about your plan regarding this baseline: what experiments from the paper are you reproducing?, for which datasets ? the more details you provide us with the better !
- [x] **Check if you are eligible for a reward**. 
    > As we have to comply with US/EU regulations, we have checked that individual contributors based on these countries or territories are eligible: Australia, Austria, Belgium, Bulgaria, Canada, Croatia, Cyprus, Czech Republic, Denmark, Estonia, Finland, France, Germany, Gibraltar, Greece, Hong Kong SAR China, Hungary, India, Ireland, Italy, Japan, Latvia, Liechtenstein, Lithuania, Luxembourg, Malta, Mexico, Netherlands, New Zealand, Norway, Poland, Portugal, Romania, Singapore, Slovakia, Slovenia, South Korea, Spain, Sweden, Switzerland, Thailand, United Arab emirates, United Kingdom United States.

    If where you are based is not on the list, please send us an email (`summer@flower.dev`) letting us know a bit about yourself (where are you currently based?, are you a university student? do you work at a public institution?). Please tell us the baselines you are interested in implementing (i.e. tell us your GitHub issue if you have crated one). We will reach back to you.
- [x] We will discuss with you about your contribution plan, if it sounds like a substantial enough contribution according to the Summer of Reproducibility rules (check our website [flower.dev/summer](https://flower.dev/summer/)), you'll get the OK from us to start working on your baseline!

## What happens next?

- [x] This item will be moved to the `In Progress` stage by a member of the Flower Team.

- [ ] Follow the [instructions for creating a new baseline](https://github.com/adap/flower/tree/main/baselines#contributing-a-new-baseline) which will guide you through the process step-by-step.

## Is something wrong or not clear ?
* Ask a question directly in your issue.
* Reach out to us via the [Flower Slack](https://flower.dev/join-slack/) and ask your question in the #summer-of-reproducibility channel
* Check all the details (including FAQ) in the Summer of Reproducibility website: flower.dev/summer

**Comment by WangYangfan** [comment]
I will reproduce Algorithm 1 on MNIST , CIFAR-10 , CIFAR-100 , and CINIC-10 datasets

**Comment by imchell** [comment]
Teamed up with @WangYangfan so my choice would be the same.

**Comment by jafermarq** [comment]
Hi @WangYangfan and @imchell, it's great to see you are interested in implementing FedNTD. This is the first team effort so far in the _Summer of Reproducibility_  🙌. Also, please bear in mind that we'll need to know who is our _primary contact_, who would be the one rewarded when all stages are completed. In [the FAQ](https://flower.dev/summer/#faq) when mention this. 

@WangYangfan you say "reproduce Algorithm 1", did you mean "table 1"? This would be ideal

**Comment by WangYangfan** [comment]
@jafermarq Yes, i mean "table 1", it is. @imchell will be the one rewarded, and I can be the primary contact

**Comment by jafermarq** [comment]
@imchell, when you have a second take a look at the _reward eligibility_ point in Step 2. Let me know if you'd be eligible once all conditions in the _Summer of Reproducibility_ are met.

**Comment by imchell** [comment]
Thanks @jafermarq, I meet the condition mentioned in Step 2. I'm based in the Netherlands and a university student.

**Comment by jafermarq** [comment]
@imchell , thanks for the quick reply! You and @WangYangfan are ready to start! I've ✅ the points above, see the _What happens next?_ for info on how to begin. Any questions please drop them here or in our Slack channel (preferably there)

**Comment by jafermarq** [comment]
Hi @WangYangfan and @imchell !
This is just a gentle reminder that the Flower *Summer of Reproducibility* is ending at the end of the month. With just a little more than 3 weeks to go, we are excited to see quite a few baselines well ahead in the process with [their respective PRs close to ready.](https://github.com/adap/flower/pulls?q=is%3Apr+is%3Aopen+label%3Asummer-of-reproducibility) If your PR is already on the list, great !! Please make sure the PR is linked to this issue (you just need to copy the URL of this issue somewhere in the main message of your PR). Ping me when you'd like me to take a look.

Also, make sure you keep an eye:eyes: on the **#summer-of-reproducibility** channel in the Flower Slack. I’ll announce very soon a new (the third!) round of 1:1 ask-me-anything sessions to help _Summer of Reproducibility_ contributors like yourself to meet the deadline. Please consider booking a time slot if you want to chat with me about your baseline, potential issues you have making your code run, how to open a PR, doubts about what to include in your readme, how to use Hydra configs more effective, etc … all questions are welcome!!

**Comment by eigengravy** [comment]
Hey, I am interested in working on this baseline. I have already completed the FedNTD implementation, will have a go at reproducing some results from the paper over the next couple of weeks.

**Comment by jafermarq** [comment]
Hi @eigengravy, great! Ping me when you'd like me to take a look at your PR. Please bear in mind the the _Flower Summer of Reproducibility_ finished some months ago. Please reach us to us either here or via [our Slack workspace](https://flower.ai/join-slack/) for longer discussions if you have some questions!

**Comment by WilliamLindskog** [comment]
Hi @eigengravy, 

Just checking in here. Are you still eager to complete this baseline implementation? 

Best regards
William

**Comment by WilliamLindskog** [comment]
Closing this for now as it is stale.
